### PR TITLE
Update azure_rm_storageaccount.py

### LINKED ISSUE #325: azure_rm_storageaccount allow_blob_public_access won't update when used on new storage account
##### SUMMARY
on new storage accounts allow_blob_public_access is null and cannot be updated to yes or no using this module

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure.azcollection.azure_rm_storageaccount won't perform an update if the allow_blob_public_access is None:

        if self.allow_blob_public_access is not None:
            if bool(self.allow_blob_public_access) != bool(self.account_dict.get('allow_blob_public_access')):
                self.results['changed'] = True
                self.account_dict['allow_blob_public_access'] = self.allow_blob_public_access
                if not self.check_mode:
                    try:
                        parameters = self.storage_models.StorageAccountUpdateParameters(allow_blob_public_access=self.allow_blob_public_access)
                        self.storage_client.storage_accounts.update(self.resource_group,
                                                                  self.name,
                                                                  parameters)
                    except Exception as exc:
                        self.fail("Failed to update account type: {0}".format(str(exc)))

##### ANSIBLE VERSION

```
ansible 2.10.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/ansible/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /var/lib/ansible/venv/dco-venv.py3/lib64/python3.6/site-packages/ansible
  executable location = /var/lib/ansible/venv/dco-venv.py3/bin/ansible
  python version = 3.6.8 (default, May  6 2020, 12:04:35) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]

```

##### CONFIGURATION

```

```

##### OS / ENVIRONMENT

Linux css-ansapp1-d1 3.10.0-1160.2.2.el7.x86_64 #1 SMP Sat Oct 17 05:06:47 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

Red Hat Enterprise Linux Server release 7.9 (Maipo)

##### STEPS TO REPRODUCE

```yaml
- name: Azure | disable blob public access on everything in {{ rg_name }}
  azure.azcollection.azure_rm_storageaccount:
    resource_group: "{{ rg_name }}"
    name: "{{ item.name }}"
    allow_blob_public_access: False
    client_id: "{{ azure.client_id }}"
    secret: "{{ azure.secret }}"
    tenant: "{{ azure.tenant }}"
    subscription_id: "{{ azure.subscription_id }}"
  when: item.allow_blob_public_access != false

```

##### EXPECTED RESULTS

{
    "changed": true,
    "state": {
        "id": "xxxxx",
        "name": "xxx",
        "location": "northcentralus",
        "resource_group": "xxxx",
        "type": "Microsoft.Storage/storageAccounts",
        "access_tier": null,
        "sku_tier": "Standard",
        "sku_name": "Standard_LRS",
        "provisioning_state": "Succeeded",
        "secondary_location": null,
        "status_of_primary": "available",
        "status_of_secondary": null,
        "primary_location": "northcentralus",
        "https_only": false,
        "minimum_tls_version": null,
        "allow_blob_public_access": null,
        "network_acls": {
            "bypass": "AzureServices",
            "default_action": "Allow",
            "virtual_network_rules": [],
            "ip_rules": []
        },
        "custom_domain": null,
        "primary_endpoints": {
            "blob": "https://xxx.blob.core.windows.net/",
            "queue": "https://xxx.queue.core.windows.net/",
            "table": "https://xxx.table.core.windows.net/"
        },
        "secondary_endpoints": null,
        "tags": {}
    },
    "invocation": {
        "module_args": {
            "resource_group": "xxxxx",
            "name": "xxxx",
            "allow_blob_public_access": false,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "subscription_id": "aaaaaaaa-bbbb-xxxx-yyyy-13e62483bb6d",
            "auth_source": "auto",
            "cloud_environment": "AzureCloud",
            "api_profile": "latest",
            "append_tags": true,
            "state": "present",
            "force_delete_nonempty": false,
            "kind": "Storage",
            "profile": null,
            "ad_user": null,
            "password": null,
            "cert_validation_mode": null,
            "adfs_authority_url": null,
            "tags": null,
            "account_type": null,
            "custom_domain": null,
            "location": null,
            "access_tier": null,
            "https_only": null,
            "minimum_tls_version": null,
            "network_acls": null,
            "blob_cors": null
        }
    },
    "_ansible_no_log": false
}
##### ACTUAL RESULTS

{
    "changed": false,
    "state": {
        "id": "xxxxxx",
        "name": "xxxx",
        "location": "northcentralus",
        "resource_group": "xxxxx",
        "type": "Microsoft.Storage/storageAccounts",
        "access_tier": null,
        "sku_tier": "Standard",
        "sku_name": "Standard_LRS",
        "provisioning_state": "Succeeded",
        "secondary_location": null,
        "status_of_primary": "available",
        "status_of_secondary": null,
        "primary_location": "northcentralus",
        "https_only": false,
        "minimum_tls_version": null,
        "allow_blob_public_access": null,
        "network_acls": {
            "bypass": "AzureServices",
            "default_action": "Allow",
            "virtual_network_rules": [],
            "ip_rules": []
        },
        "custom_domain": null,
        "primary_endpoints": {
            "blob": "https://xxxxx.blob.core.windows.net/",
            "queue": "https://xxxx.queue.core.windows.net/",
            "table": "https://xxxx.table.core.windows.net/"
        },
        "secondary_endpoints": null,
        "tags": {}
    },
    "invocation": {
        "module_args": {
            "resource_group": "xxxxxxx",
            "name": "csstestvm3d16800",
            "allow_blob_public_access": false,
            "client_id": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "secret": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "tenant": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "subscription_id": "aaaaaaaa-bbbb-xxxx-yyyy-13e62483bb6d",
            "auth_source": "auto",
            "cloud_environment": "AzureCloud",
            "api_profile": "latest",
            "append_tags": true,
            "state": "present",
            "force_delete_nonempty": false,
            "kind": "Storage",
            "profile": null,
            "ad_user": null,
            "password": null,
            "cert_validation_mode": null,
            "adfs_authority_url": null,
            "tags": null,
            "account_type": null,
            "custom_domain": null,
            "location": null,
            "access_tier": null,
            "https_only": null,
            "minimum_tls_version": null,
            "network_acls": null,
            "blob_cors": null
        }
    },
    "_ansible_no_log": false
}

```paste below
ok: [localhost] => {"changed": false, "state": {"access_tier": null, "allow_blob_public_access": null, "custom_domain": null, "https_only": false, "id": "/subscriptions/aaaaaaaa-bbbb-xxxx-yyyy-13e62483bb6d/resourceGroups/rg_nam_ss_ncus_workshopdemo2/providers/Microsoft.Storage/storageAccounts/xxxxx", "location": "northcentralus", "minimum_tls_version": null, "name": "xxxxx", "network_acls": {"bypass": "AzureServices", "default_action": "Allow", "ip_rules": [], "virtual_network_rules": []}, "primary_endpoints": {"blob": "https://xxxxx.blob.core.windows.net/", "queue": "https://xxxx.queue.core.windows.net/", "table": "https://cxxxx.table.core.windows.net/"}, "primary_location": "northcentralus", "provisioning_state": "Succeeded", "resource_group": "xxxxx", "secondary_endpoints": null, "secondary_location": null, "sku_name": "Standard_LRS", "sku_tier": "Standard", "status_of_primary": "available", "status_of_secondary": null, "tags": {}, "type": "Microsoft.Storage/storageAccounts"}}
```

### LINKED ISSUE #457: azure_rm_storageaccount module not passing https_only true on account creation
##### SUMMARY
In the azure_rm_storageaccount module, the 'https_only' flag value always seems to pass 'false' to Azure when run, even when set to true or yes.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure_rm_storageaccount

##### ANSIBLE VERSION
```
ansible 2.9.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/apps/ansible/library']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Aug  7 2019, 00:51:29) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```
And
```
Ansible 2.10.X
```
Collection version:
```
$ ansible-galaxy collection list

# /home/XXXX/.ansible/collections/ansible_collections
Collection         Version
------------------ -------
azure.azcollection 1.4.0  
```

##### CONFIGURATION
```
CACHE_PLUGIN(/etc/ansible/ansible.cfg) = redis
CACHE_PLUGIN_CONNECTION(/etc/ansible/ansible.cfg) = localhost:6379:0
COMMAND_WARNINGS(/etc/ansible/ansible.cfg) = False
DEFAULT_FORKS(/etc/ansible/ansible.cfg) = 50
DEFAULT_GATHERING(/etc/ansible/ansible.cfg) = smart
DEFAULT_HOST_LIST(/etc/ansible/ansible.cfg) = [u'/apps/ansible/inventories/dev/hosts']
DEFAULT_LOG_PATH(/etc/ansible/ansible.cfg) = /apps/ansible/log/ansible.log
DEFAULT_MODULE_PATH(/etc/ansible/ansible.cfg) = [u'/apps/ansible/library']
DEFAULT_MODULE_UTILS_PATH(/etc/ansible/ansible.cfg) = [u'/apps/ansible/module_utils']
DEFAULT_ROLES_PATH(/etc/ansible/ansible.cfg) = [u'/apps/ansible/roles']
```

##### OS / ENVIRONMENT
Target is just the Azure cloud - storage accounts in Azure.

##### STEPS TO REPRODUCE
You use the code below, first pass use create_account function, doesn't push the https_only to azure, if you run a second time the update_account function is used and the option is set on the storage account.

Code sample to reproduce:
```
---
- name: Create RG and secure storage account
  hosts: localhost
  gather_facts: false
  collections:
    - azure.azcollection

  tasks:
  - name: Create RG
    azure.azcollection.azure_rm_resourcegroup:
      name: "{{ item }}"
      location: canadacentral
    loop:
      - rg01

  - name: Create storage account true
    azure.azcollection.azure_rm_storageaccount:
      resource_group: rg01
      https_only: true
      name: "{{ item }}"
      kind: StorageV2
      type: Standard_LRS
      allow_blob_public_access: no
    loop:
      - pp9901
```

##### EXPECTED RESULTS
Expected to have the param set for the module match the correct value being passed to Azure and to be able to create storage accounts with the https_only flag set to true since that feature is available in Azure and the Ansible module documents that it should be working.

##### ACTUAL RESULTS
Ansible-playbook output using -vvv. You can see the differences between the invocation and the state.
```
[...]
using module file /home/XXXX/.ansible/collections/ansible_collections/azure/azcollection/plugins/modules/azure_rm_storageaccount.py
[...]
changed: [localhost] => (item=pp9901) => {
    "ansible_loop_var": "item",
    "changed": true,
    "invocation": {
        "module_args": {
            "access_tier": null,
            "account_type": "Standard_LRS",
            "ad_user": null,
            "adfs_authority_url": null,
            "allow_blob_public_access": false,
            "api_profile": "latest",
            "append_tags": true,
            "auth_source": "auto",
            "blob_cors": null,
            "cert_validation_mode": null,
            "client_id": null,
            "cloud_environment": "AzureCloud",
            "custom_domain": null,
            "force_delete_nonempty": false,
            "https_only": true,
            "kind": "StorageV2",
            "location": null,
            "minimum_tls_version": null,
            "name": "pp9901",
            "network_acls": null,
            "password": null,
            "profile": null,
            "resource_group": "rg01",
            "secret": null,
            "state": "present",
            "subscription_id": null,
            "tags": null,
            "tenant": null,
            "type": "Standard_LRS"
        }
    },
    "item": "pp9901",
    "state": {
        "access_tier": "Hot",
        "allow_blob_public_access": false,
        "custom_domain": null,
        "https_only": false,
        "id": "/subscriptions/XXXXXXXXXXXXX/resourceGroups/rg01/providers/Microsoft.Storage/storageAccounts/pp9901",
        "location": "canadacentral",
        "minimum_tls_version": null,
        "name": "pp9901",
        "network_acls": {
            "bypass": "AzureServices",
            "default_action": "Allow",
            "ip_rules": [],
            "virtual_network_rules": []
        },
        "primary_endpoints": {
            "blob": "https://pp9901.blob.core.windows.net/",
            "queue": "https://pp9901.queue.core.windows.net/",
            "table": "https://pp9901.table.core.windows.net/"
        },
        "primary_location": "canadacentral",
        "provisioning_state": "Succeeded",
        "resource_group": "rg01",
        "secondary_endpoints": null,
        "secondary_location": null,
        "sku_name": "Standard_LRS",
        "sku_tier": "Standard",
        "status_of_primary": "available",
        "status_of_secondary": null,
        "tags": null,
        "type": "Microsoft.Storage/storageAccounts"
    }
}
```
As you can see the module_args show clearly that ""https_only": true" is set and we can see in the state that it's not set ""https_only": false"

Azure activity log output confirm the problem is not on azure side, azure receive supportsHttpsTrafficOnly to false:
```
    "properties": {
        "requestbody": "{\"sku\":{\"name\":\"Standard_LRS\",\"tier\":\"Standard\"},\"kind\":\"StorageV2\",\"location\":\"canadacentral\",\"properties\":{\"supportsHttpsTrafficOnly\":false,\"allowBlobPublicAccess\":false}}",
        "eventCategory": "Administrative",
        "entity": "/subscriptions/XXXXXXXXXXXXXXXXX/resourceGroups/rg01/providers/Microsoft.Storage/storageAccounts/pp9901",
        "message": "Microsoft.Storage/storageAccounts/write",
    },
```

The problem is in the logic of the create_account function. I fixed it with a quick and dirty fix here: https://github.com/arsenicks/azure/commit/4ac4b9e5e2be40f85ce530f4e562ed7b422ea790

Not to criticize the work of someone but, I don't understand why this exist in the code in the given context:
```
        if bool(self.https_only):
            self.https_only = False
```
This force the https_only flag to false if the bool is set right ? I would be more than happy to submit a clean fix here and push it thru but @paultaiton has stated in https://github.com/ansible-collections/azure/issues/325#issuecomment-731650897 in NÉov 2020 he's already working on this module so I don't want to complicate things but this is a real problem for us.

Related issue which has been closed: https://github.com/ansible-collections/azure/issues/184
Related issue which seems to come from the same logic problem: https://github.com/ansible-collections/azure/issues/325

Let me know if you'd like me to contribute or if @paultaiton is still working on this.

Thanks.

### PR DESCRIPTION
**Pull request by paultaiton**
##### SUMMARY

Fix passing minimum_tls_version, allow_blob_public_access, and https_only on new storage account creation.

Fixes #457 
Fixes #325 
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

azure_rm_storageaccount

##### ADDITIONAL INFORMATION

This is a bugfix in the creation of new storage account, and update of tests to catch this bug in future regressions.

```paste below

```

### CODE REVIEW

**Comment by paultaiton** [comment]
@Fred-sun,
I believe you wrote the original code that I'm modifying in this PR. Your review would be appreciated.

@arsenicks,
This should correct the problem you have in the short term, until I can complete the more comprehensive PR.

**Comment by arsenicks** [comment]
@paultaiton Ohhh thanks a lot for this!!

**Review by arsenicks — DISMISSED** [review]

File: plugins/modules/azure_rm_storageaccount.py [hunk 1]
```diff
@@ -823,7 +823,7 @@ def create_account(self):
             if self.blob_cors:
                 account_dict['blob_cors'] = self.blob_cors
             return account_dict
-        if bool(self.https_only):
+        if not bool(self.https_only):
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_storageaccount.py]:
```suggestion
        if self.https_only is None:
```

File: plugins/modules/azure_rm_storageaccount.py [hunk 2]
```diff
@@ -823,7 +823,7 @@ def create_account(self):
...
             self.https_only = False
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_storageaccount.py]:
```suggestion
            self.https_only = True
```

**Review by Fred-sun — COMMENTED** [review]
@paultaiton I have checked the parameter in azure-cli, https-only default value is true. Please update follow the comment and update the doc---line 90. Thank you very much!

https://github.com/ansible-collections/azure/blob/bb56aec4b2c8ecd71ebb8ab0267f575967b8dedd/plugins/modules/azure_rm_storageaccount.py#L90
```e.g.
link: https://docs.microsoft.com/en-us/cli/azure/storage/account?view=azure-cli-latest#az_storage_account_create

**Comment by paultaiton** [comment]
If we are setting https_only parameter to be True by default (which makes sense to match storageaccount API updates 2019-06-01,) then the line 826 
```
         if not bool(self.https_only): 
             self.https_only = False 
``` 

is no longer necessary, since the property can only ever be True or False, and never None.

I've modified the code to match this convention instead.

**Comment by Fred-sun** [comment]
> If we are setting https_only parameter to be True by default (which makes sense to match storageaccount API updates 2019-06-01,) then the line 826
> 
> ```
>          if not bool(self.https_only): 
>              self.https_only = False 
> ```
> 
> is no longer necessary, since the property can only ever be True or False, and never None.
> 
> I've modified the code to match this convention instead.

Yes! agree with your comments!

File: plugins/modules/azure_rm_storageaccount.py [hunk 3]
```diff
@@ -87,7 +87,7 @@
         description:
             - Allows https traffic only to storage service when set to C(true).
             - Allows update storage account property when set to C(False).
-            - Default value is C(False).
+            - Default value is C(True).
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_storageaccount.py]:
```suggestion
        default: True
```

**Review by Fred-sun — COMMENTED** [review]
Please

**Comment by paultaiton** [comment]
@Fred-sun 
I added a couple more changes since I noticed that minimum_tls_version had the same bug, and to make blob_public_access more explicit and obvious with the defaults.

File: plugins/modules/azure_rm_storageaccount.py [hunk 4]
```diff
@@ -87,23 +87,23 @@
...
             - TLS1_2
+        default: 'TLS1_0'
         version_added: "1.0.0"
     allow_blob_public_access:
         description:
             - Allows blob containers in account to be set for anonymous public access.
             - If set to false, no containers in this account will be able to allow anonymous public access.
-            - Default value is C(True).
         type: bool
+        default: True
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_storageaccount.py]: not need! The default value for this property is null

**Review by Fred-sun — COMMENTED** [review]
Please!

**Comment by Fred-sun** [comment]
@paultaiton Please update the follow lines,  If set minimum_tls_version default value!, It will update the minimum_tls_version value when update account blob public access.  advice change to '- output.storageaccounts[0].minimum_tls_version == "TLS1_0" '
https://github.com/ansible-collections/azure/blob/553cbd277fb84488fa11a1a057efdfc44b99412b/tests/integration/targets/azure_rm_storageaccount/tasks/main.yml#L190

**Comment by arsenicks** [comment]
> @Fred-sun
> I added a couple more changes since I noticed that minimum_tls_version had the same bug, and to make blob_public_access more explicit and obvious with the defaults.

I was aware of it and can confirm it's not working as expected in 1.4 but didn't want to bother you with this in the short term as you were working on the whole thing :P Anyway thanks for fixing it, let me know if you need any kind of test. We have a policy in place that was blocking the creation because of the default ACL not being Deny, my first debug lead me to think this is not the same problem but if you can take a second to lookup that part of the code and see if it's similar that would be great.

I'll create another issue for the ACL when I have more information, it's not clear what cause the problem right now.

One question for you guy's, not putting any pressure, just want to know how released are managed with the collection. Do you ship a new collection version on a specific "schedule" or just as needed ? If so is there a planned released in the coming weeks ?

Thanks a lot to you both!

**Comment by Fred-sun** [comment]
@arsenicks There will probably be an edition of Collections in a month or two.

**Comment by Fred-sun** [comment]
@haiyuazhang

File: plugins/modules/azure_rm_storageaccount.py [hunk 5]
```diff
@@ -822,12 +821,6 @@ def create_account(self):
             if self.blob_cors:
                 account_dict['blob_cors'] = self.blob_cors
             return account_dict
-        if bool(self.https_only):
-            self.https_only = False
-        if bool(self.minimum_tls_version):
-            self.minimum_tls_version = 'TLS1_0'
-        if bool(self.allow_blob_public_access):
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_storageaccount.py:829]: if self.allow_blob_public_access is None:
            self.allow_blob_public_access = True

File: plugins/modules/azure_rm_storageaccount.py [hunk 6]
```diff
@@ -822,12 +821,6 @@ def create_account(self):
             if self.blob_cors:
                 account_dict['blob_cors'] = self.blob_cors
             return account_dict
-        if bool(self.https_only):
-            self.https_only = False
-        if bool(self.minimum_tls_version):
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_storageaccount.py:827]: if self.minimum_tls_version is None:
            self.minimum_tls_version = 'TLS1_0'

**Review by Fred-sun — COMMENTED** [review]
Please!

**Comment by paultaiton** [comment]
Hey @Fred-sun ,
The test that failed is correct, it uncovered a bug I introduced.
Luckily I finally figured out how to get these tests running locally, so I'm updating both the integration tests and my code to account for omitting parameters while updating existing accounts. Please hold off on reviewing anything until I comment again, as I've corrected many of the bugs that haven't been pushed up yet.

**Comment by paultaiton** [comment]
@Fred-sun 

OK, I have reorganized and updated the tests to include more checks that should catch similar bugs in the future. It now creates and updates two storage accounts using implicit defaults and explicit settings for creating new accounts and updating existing accounts.

I added to documentation to explain what happens during parameter omission, but left it to use the API (also matches CLI) behavior.

I've run the integration tests and everything was successful, so I think it's ready for review for everything within the scope of this immediate fix. I did uncover a problem with the returned values for network_acls and blob_cors (integration test lines 227-229 that are commented out,) but it's outside the scope of the code I've touched, and I'm not familiar enough with its operation to fix it within this PR. I'll try and get to it later if I can.

Appreciate your help Fred.

**Comment by l3ender** [comment]
@paultaiton Would you mind sharing detail on how you were able to get tests running locally? I've been trying to do the same but haven't had luck. Any detail you can share on running specific tests and the test suite would be appreciated!

**Comment by paultaiton** [comment]
> @paultaiton Would you mind sharing detail on how you were able to get tests running locally? I've been trying to do the same but haven't had luck. Any detail you can share on running specific tests and the test suite would be appreciated!

Yes, absolutely. I was already planning on making a Documentation PR that describes exactly how to set up an environment for linting and integration tests.

File: plugins/modules/azure_rm_storageaccount.py [hunk 7]
```diff
@@ -706,8 +705,7 @@ def update_account(self):
                 except Exception as exc:
                     self.fail("Failed to update account type: {0}".format(str(exc)))
 
-        if self.allow_blob_public_access is not None:
-            if bool(self.allow_blob_public_access) != bool(self.account_dict.get('allow_blob_public_access')):
+        if self.allow_blob_public_access is not None and self.allow_blob_public_access != self.account_dict.get('allow_blob_public_access'):
                 self.results['changed'] = True
```
Comment (Fred-sun) [inline plugins/modules/azure_rm_storageaccount.py]: @paultaiton  over indented! Thanks!

Comment (paultaiton) [reply plugins/modules/azure_rm_storageaccount.py]: Got it @Fred-sun . Somehow my linter got disabled. Thanks.

**Comment by paultaiton** [comment]
Thanks again for your help @haiyuazhang and @Fred-sun . Your support is appreciated.

@arsenicks please let me know if you see any trouble after switching to this new module update.

**Comment by l3ender** [comment]
@paultaiton Is there any detail you can share in the meantime on how to set up an environment for linting and integration tests? Being able to lint/test on my local environment would help tremendously. Once getting it set up, I'm happy to create and/or contribute to a documentation PR.

**Comment by paultaiton** [comment]
@l3ender 

I use pycodestyle in vscode with --max-line-length=160       and
--ignore=E###,W###

As command line arguments (settings in python extension.) I forget the E number, but the one for module imports not at top of file, and the warn for operators at start of new line. The latter is acceptable according to PEP8, so I consider it a fault of the pycodestyle lint.

The ansible-test bit I'll have to get you when I'm back home, as there is a cloud config file you need to define that I can't recall from memory.

**Pull request merged**